packages/gatsby/src/utils/flags.ts

```diff
@@ -55,16 +96,67 @@ const activeFlags: Array<IFlag> = [
     experimental: false,
     description: `Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.`,
     umbrellaIssue: `https://gatsby.dev/query-on-demand-feedback`,
-    noCi: true,
+    noCI: true,
+    testFitness: (): fitnessEnum => {
+      // Take a 10% of slice of users.
+      if (sampleSiteForExperiment(`QUERY_ON_DEMAND`, 10)) {
+        let isPluginSharpNewEnoughOrNotInstalled = false
+        try {
+          // Try requiring plugin-sharp so we know if it's installed or not.
+          // If it does, we also check if it's new enough.
+          // eslint-disable-next-line
+          require(`gatsby-plugin-sharp`).version
```

Comment:
```suggestion
          require.resolve(`gatsby-plugin-sharp`)
```
We don't need to actually load the module - resolving will also throw if not found

Reply:
oh right — I was using the version but then didn't

Reply:
hehe, but that wouldn't work either - because `package.json` is not imported and `version` is not export in gatsby-plugin-sharp/index :)

Reply:
heh but it did error when it wasn't found!!!